src/shared/types/routes.ts

```diff
@@ -0,0 +1,3 @@
+export type ProtectedProps= {
+  children: JSX.Element;
```

Comment:
@BardhMustafa I see you are in sync with Rron :D I like your idea of putting all types to a folder, but I think is more convenient to have them close to component file, the reason is there are easier to maintain, no need for export (since most of the times are being used to describe props) and easier to remove legacy when needed.

I would suggest to add the props here in the same file with ProtectedProps